### DISCUSSION TITLE
Learning

### DISCUSSION THREAD

**Discussion by BenasB** [vu-vibedosa/apsitvarkom-be#1] [General]
Back end:
- [Web API tutorial](https://docs.microsoft.com/en-us/aspnet/core/tutorials/first-web-api?view=aspnetcore-6.0&tabs=visual-studio)
- [Asynchronous programming in C#](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/concepts/async/)
- [Backend web development - a complete overview (2021)](https://youtu.be/XBu54nfzxAQ)

Front end:
- [React Beta docs](https://beta.reactjs.org/learn) - official stable docs are somewhat outdated, learn with the beta version
- [Ben Awad's React + Typescript video](https://www.youtube.com/watch?v=Z5iWr6Srsj8&ab_channel=BenAwad) - after checking out react basics, this is the go to place of quick learning of how to use React with TS
- [Official Typescript handbook](https://www.typescriptlang.org/docs/handbook/2/basic-types.html) - might be a bit hard to understand at the start
- [Tailwindcss core concepts](https://tailwindcss.com/docs/utility-first)

**Comment by BenasB** [comment]
Feel free to add in any good resources that helped you. I think you should be able to edit my post

**Comment by 6ar8nas** [comment]
Personally found out about [this](https://www.meziantou.net/visual-studio-tips-and-tricks-paste-as-json.htm) just now. Maybe someone will also find it interesting or useful.

Our agreed on PollutedLocation JSON object for testing purposes: 
```json
{
    "id": "5be2354e-2500-4289-bbe2-66210592e17f",
    "location": {
        "longitude": -78.948237,
        "latitude": 35.929673
    },
    "radius": 10,
    "severity": "Moderate",
    "spotted": "2022-09-14T17:35:23Z",
    "progress": 0,
    "notes": "Lorem ipsum"
}
```
